src/api/common_api.js

```diff
@@ -309,52 +309,113 @@ module.exports = {
                 Statement: {
                     type: 'array',
                     items: {
-                        type: 'object',
-                        required: ['Effect', 'Action', 'Principal', 'Resource'],
-                        properties: {
-                            Sid: {
-                                type: 'string'
-                            },
-                            Effect: {
-                                enum: ['Allow', 'Deny'],
-                                type: 'string'
-                            },
-                            Action: {
-                                $ref: '#/definitions/string_or_string_array'
+                        allOf: [
+                            {
+                                oneOf: [
```

Comment:
Not an expert with this but can't we combine the 2 parts? the principle with the principles in the bottom and all the others (Resource and Action)? This looks really unreadable...
also, leave the type and required in the beginning if you can so at least it will look like something we know from the other APIs